tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+/**
+ * ownCloud - mediametadata
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Jalpreet Singh Nanda (:imjalpreet) <jalpreetnanda@gmail.com>
+ * @copyright Jalpreet Singh Nanda (:imjalpreet) 2016
+ */
+
+namespace OCA\MediaMetadata\Hooks;
+
+
+use OCA\MediaMetadata\AppInfo\Application;
+use OCA\MediaMetadata\Tests\TestCase;
+
+/**
+ * Class ImageHooksTest
+ *
+ * @package OCA\MediaMetadata\Hooks
+ */
+class ImageHooksTest extends TestCase {
+	/**
+	 * @var \OCA\MediaMetadata\Hooks\ImageHooks
+	 */
+	protected $imageHooks;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OC\Files\Node\Root
+	 */
+	protected $root;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ImageDimensionMapper
+	 */
+	protected $mapper;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCP\IDb
+	 */
+	protected $db;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\ExtractMetadata
+	 */
+	protected $extractor;
+	/**
+	 * @var \PHPUnit_Framework_MockObject_MockObject|\OCA\MediaMetadata\Services\StoreMetadata
+	 */
+	protected $dbManager;
+
+	protected function setUp() {
+		parent::setUp();
+
+		$app = new Application();
+		$container = $app->getContainer();
+
+		$this->root = $this->getMockBuilder('OC\Files\Node\Root')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->db = $this->getMockBuilder(
+			'\OCP\IDBConnection')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->mapper = $this->getMockBuilder('OCA\MediaMetadata\Services\ImageDimensionMapper')
+			->setConstructorArgs([
+				$this->db,
+			])
+			->getMock();
+		$this->extractor = $this->getMockBuilder(
+			'OCA\MediaMetadata\Services\ExtractMetadata')
+			->disableOriginalConstructor()
+			->getMock();
+		$this->dbManager = $this->getMockBuilder(
+			'\OCA\MediaMetadata\Services\StoreMetadata')
+			->setConstructorArgs([
+				$this->mapper
+			])
+			->getMock();
+		$this->imageHooks = new ImageHooks(
+			$this->root,
+			$this->mapper,
+			$this->extractor,
+			$this->dbManager,
+			$container->query('ServerContainer')->getConfig()->getSystemValue('datadirectory')
+		);
+	}
+
+	public function testPostCreate() {
+		$location = '/testFolder/test.jpg';
+		$fileId = 260495;
+		$jpgFile = $this->mockJpgFile($location, $fileId);
+
+		$app = new Application();
+		$container = $app->getContainer();
+
+		$absolutePath = $container->query('ServerContainer')->getConfig()->getSystemValue('datadirectory').$location;
+
+		$this->assertEquals($jpgFile->getPath(), $location);
+
+		$this->extractor->expects($this->once())
+			->method('extract')
+			->with($absolutePath);
+
+		$metadata = null;
+
+		$this->dbManager->expects($this->once())
+			->method('store')
+			->with(
+				$metadata,
+				$jpgFile
+			)
+			->willReturn(true);
+
+		$this->imageHooks->postCreate($jpgFile);
```

Comment:
I think we need to modify postCreate to return the "result". That way you can test the output.
Maybe use an if-else block and log the error if "result" is false?